logging/logging.go

```diff
@@ -185,3 +185,27 @@ func (l *NoOpLogger) SetLevel(level Level) {
 func (l *NoOpLogger) GetLevel() Level {
 	return l.level
 }
+
+type requestContextKey string
+
+// ReqCtxKey is the key used to access the request context.
+const ReqCtxKey = requestContextKey("request-context-key")
```

Comment:
Exporting this is defeating the purpose here: It's done via non-exported type to ensure that we control how it's set and read.

Alternatively, we could keep `reqCtxKey` unexported and expose a method like
```go
func WithRequestID(ctx context.Context, reqID string) context.Context
```
that injects the request ID accordingly, and
```go
func RequestIDFromContext(context.Context) string
```
or something like that for retrieval.

Reply:
Done.